src/Umbraco.Infrastructure/Members/UmbracoMembersUserManager.cs

```diff
@@ -0,0 +1,244 @@
+                {
+                    return result == UmbracoMembersUserPasswordCheckerResult.ValidCredentials;
+                }
+            }
+
+            // we cannot proceed if the user passed in does not have an identity
+            if (member.HasIdentity == false)
+            {
+                return false;
+            }
+
+            // use the default behavior
+            return await base.CheckPasswordAsync(member, password);
+        }
+
+        /// TODO: from BackOfficeUserManager duplicated, could be shared
+        /// <summary>
+        /// This is copied from the underlying .NET base class since they decided to not expose it
+        /// </summary>
+        /// <param name="user">The user to update the security stamp for</param>
+        /// <returns>Task returns</returns>
+        private async Task UpdateSecurityStampInternal(T user)
+        {
+            if (SupportsUserSecurityStamp == false)
+            {
+                return;
+            }
+
+            await GetSecurityStore().SetSecurityStampAsync(user, NewSecurityStamp(), CancellationToken.None);
+        }
+
+        /// TODO: from BackOfficeUserManager duplicated, could be shared
+        /// <summary>
+        /// This is copied from the underlying .NET base class since they decided to not expose it
+        /// </summary>
+        /// <returns>Return a user security stamp</returns>
+        private IUserSecurityStampStore<T> GetSecurityStore()
+        {
+            if (!(Store is IUserSecurityStampStore<T> store))
+            {
+                throw new NotSupportedException("The current user store does not implement " + typeof(IUserSecurityStampStore<>));
+            }
+
+            return store;
+        }
+
+        /// TODO: from BackOfficeUserManager duplicated, could be shared
+        /// <summary>
+        /// This is copied from the underlying .NET base class since they decided to not expose it
+        /// </summary>
+        /// <returns>Returns a new security stamp</returns>
+        private static string NewSecurityStamp() => Guid.NewGuid().ToString();
+
+        /// <summary>
+        /// TODO: from BackOfficeUserManager duplicated, could be shared
+        /// Helper method to generate a password for a member based on the current password validator
+        /// </summary>
+        /// <returns>The generated password</returns>
+        public string GeneratePassword()
+        {
+            _passwordGenerator ??= new PasswordGenerator(PasswordConfiguration);
+            string password = _passwordGenerator.GeneratePassword();
+            return password;
+        }
+
+        /// <summary>
+        /// Helper method to validate a password based on the current password validator
+        /// </summary>
+        /// <param name="password">The password to update</param>
+        /// <returns>The validated password</returns>
+        public async Task<List<IdentityResult>> ValidatePassword(string password)
```

Comment:
**question:** this one already exists on the UserManager, albeit with a slightly different signature. Do we need this version?